src/Microsoft.DotNet.XHarness.Android/AdbRunner.cs

```diff
@@ -423,28 +453,25 @@ public List<string> PullFiles(string devicePath, string localPath)
             catch (Exception toLog)
             {
                 logger.LogError(toLog, $"Exception thrown while trying to find compatible device with {propertyName} {apkRequiredProperty}");
-                return null;
+                return new Dictionary<string, string>();
             }
 
             if (allDevicesAndTheirProperties.Count == 0)
             {
                 logger.LogError("No attached device detected");
-                return null;
+                return new Dictionary<string, string>();
             }
 
-            if (allDevicesAndTheirProperties.Any(kvp => apkRequiredProperty.Equals(kvp.Value, StringComparison.OrdinalIgnoreCase)))
-            {
-                // Key-value tuples here are of the form <device serial number, device property>
-                KeyValuePair<string, string?> firstAvailableCompatible = allDevicesAndTheirProperties.FirstOrDefault(kvp => apkRequiredProperty.Equals(kvp.Value, StringComparison.OrdinalIgnoreCase));
-                logger.LogDebug($"Using first-found compatible device of {allDevicesAndTheirProperties.Count} total- serial: '{firstAvailableCompatible.Key}' - {propertyName}: {firstAvailableCompatible.Value}");
-                return firstAvailableCompatible.Key;
-            }
-            else
+            var result = allDevicesAndTheirProperties
+                .Where(kvp => !string.IsNullOrEmpty(kvp.Value) && kvp.Value.Split().Contains(apkRequiredProperty))

```

Comment:
What characters are we splitting here by?
(I don't know what the method does called like this, so asking)

Can it happen that the default set of splitters contains a character that is also in the `apkRequiredProperty`? Because if for example a '-' or ':' was there, it wouldn't work

Reply:
For the list of the installed app for example. It is splitting by whitespace-like characters, isn't it? At least I check for arm64-v8a arch and it wasn't splitted that way

Reply:
Yup, should be good then